internal/cmd/gocdk/build.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/xerrors"
+)
+
+func build(ctx context.Context, pctx *processContext, args []string) error {
+	f := flag.NewFlagSet("build", flag.ContinueOnError)
+	list := f.Bool("list", false, "display Docker images of this project")
+	if err := f.Parse(args); xerrors.Is(err, flag.ErrHelp) {
+		return nil
+	} else if err != nil {
+		return usagef("gocdk build: %w", err)
+	}
+	if *list {
+		return listBuilds(ctx, pctx, f.Args())
+	}
+	return xerrors.New("not implemented")
+}
+
+func listBuilds(ctx context.Context, pctx *processContext, args []string) error {
+	if len(args) != 0 {
+		return usagef("gocdk build --list")
+	}
+	moduleRoot, err := findModuleRoot(ctx, pctx.workdir)
+	if err != nil {
+		return xerrors.Errorf("list builds: %w", err)
+	}
+	imageName, err := moduleDockerImageName(moduleRoot)
+	if err != nil {
+		return xerrors.Errorf("list builds: %w", err)
+	}
+	c := exec.CommandContext(ctx, "docker", "images", imageName)
```

Comment:
Is there a way to interact w/ docker via a client API? That could be easier to test/mock

Reply:
There is, but bringing in a large dependency for first commit seemed a bridge too far. Opened #1804 to track.